contracts/plugins/assets/stargate/StargateRewardableWrapper.sol

```diff
@@ -36,27 +36,42 @@ contract StargateRewardableWrapper is RewardableERC20Wrapper {
         }
         require(pid != type(uint256).max, "Invalid pool");
 
-        pool_.approve(address(stakingContract_), type(uint256).max); // TODO: Change this!
-
         pool = pool_;
         poolId = pid;
         stakingContract = stakingContract_;
         stargate = stargate_;
     }
 
     function _claimAssetRewards() internal override {
-        stakingContract.deposit(poolId, 0);
+        IStargateLPStaking.PoolInfo memory poolInfo = stakingContract.poolInfo(poolId);
+
+        if (poolInfo.allocPoint != 0 && totalSupply() != 0) {
+            stakingContract.deposit(poolId, 0);
+        } else {
+            stakingContract.emergencyWithdraw(poolId);
+        }
     }
 
-    function _afterDeposit(uint256 _amount, address to) internal override {
-        require(to == msg.sender, "Only the sender can deposit");
+    function _afterDeposit(uint256, address) internal override {
+        uint256 underlyingBalance = underlying.balanceOf(address(this));
+        IStargateLPStaking.PoolInfo memory poolInfo = stakingContract.poolInfo(poolId);
 
-        stakingContract.deposit(poolId, _amount);
+        if (poolInfo.allocPoint != 0 && underlyingBalance != 0) {
```

Comment:
if the pool is disabled, should we not revert here?

Reply:
Only the staking pool is disabled, not the actual LP. So this allows people to still enter/exit an RToken if they wish to. This just makes it so we don't get the rewards if the staking pool is disabled. Everything else remains working.

Reply:
Got it. 👍